### ISSUE TITLE
Challenge : compute H2O on Radis-Lab

### ISSUE THREAD

**Issue by erwanp** [radis/radis-lab#24]
In the latest commits to RADIS-Lab https://github.com/radis/radis-lab/commit/567b0250dac691cfdedae83a79fbdbe32332a92c I added the auto-download of HITEMP H2O lines. 

Challenge is to be able to run a computation directly on RADIS-Lab ! 

Currently, we are limited because : 
1.  the .h5 cache files are not generated during the Docker [PostBuild](https://github.com/radis/radis-lab/blob/main/binder/postBuild) phase (it crashed) ; probably because of Memory error while unpacking them.  
2.  you can still start computing from Radis-Lab, but it takes ages to generate the .h5 files from the .par files. 
3.  Then, it will be a challenge to know if 8 GB RAM provided by GESIS are enough to compute an H2O spectrum. 

1. could be helped a lot by using Vaex to stream the .par files into HDF5 files (a small change of the Pandas syntax used in https://github.com/radis/radis/pull/203 and https://github.com/radis/radis/pull/281).  If .h5 are cached during Docker build; the Docker build will likely take 2 hrs ; but user computations will be instantaneous then ! ( kind reminder that we need #13  to be able to add Example Notebooks here without restarting the Docker !)
3. May work after @gagan-aryan 's Reduce Memory Usage https://github.com/radis/radis/issues/118 project

**Comment by Springder** [comment]
I failed to compute H2O on Radis-Lab(990-1000 cm-1), even if 8 GB RAM was provided by GESIS. It's really necessary to unzip all par files when I just need 990-1000 cm-1, but in fact, that's what it does, converting all files into h5 files. This may be the reason for the failure？

**Comment by erwanp** [comment]
Yes! Good news : in [0.9.30](https://github.com/radis/radis/milestone/7), we only download, unzip & parse the files needed.  This is already implemented in develop version (https://github.com/radis/radis/pull/281) and will be available on RADIS-Lab as soon as 0.9.30 is released

**Comment by Springder** [comment]
In the current version, can I upload the calculated H5 file to radis-lab for calculation? Will it work?

**Comment by ariadnc** [comment]
Hello, I tried using the HITEMP-H2O data and it is not available in the jupyter-lab environment. I also tried running it in my computer but it seams like there is a problem converting most of the hitemp molecules .par files in .h5 files. The errors you get is that is not concatenated and that there is an error on the data stream. 

Do you know if I can get the .h5 directly without having to convert it anywhere? I see that the path being path (example):
 `path: "/home/jovyan/.radisdb/hitemp/NO2-10_HITEMP2019.hdf5"`

**Comment by erwanp** [comment]
Hello @ariadnc thanks for reporting 

unfortunately Radis Lab is not maintained I'm even surprised it still works. Id suggest to run the codes on a local machine first. Do you have use cases that specifically require an online version ?

**Comment by ariadnc** [comment]
Radis Lab does not work anymore, I was in the virtual machine because the local machine did not work for the HITEMP database. The local machine does not convert the downloaded files from the HITRAN HITEMP website into the format that the library uses for analysis. This only happens with the HITEMP database, where it either does not find the .bz2 files or it just downloads a corrupted file that can not be extracted properly. But I did some changes in the code in my local machine and is solved now. 

How I went around this problem is downloading it myself from the website and add it it to the hitemp folder in my database in my local machine. That method worked for molecules like CO or OH (witch only have one file). But for both H2O and CO2 molecules witch is composed of a list of .zip files it did not even have access to them. 

I solved these issues by first adding the download url and path manually in the .json fie of the radis database. Then I downloaded the files manually from the website and added them to the database in my local maachine. And then I changed the function fetch urls and added the names of the files: 

```
 def fetch_urlnames(self):
        r"""requires connection"""

        if self.urlnames is not None:
            return self.urlnames

        molecule = self.molecule

        if molecule in ["H2O", "CO2"]:

            base_url, Ntotal_lines_expected, _, _ = self.fetch_url_Nlines_wmin_wmax()
            response = urllib.request.urlopen(base_url)
            response_string = response.read().decode()
            print(response_string)
            if molecule == "CO2":
                base_url = "https://hitran.org/files/HITEMP/HITEMP-2010/CO2_line_list/"
            
                inputfiles = [ 
                    "02_00000-00500_HITEMP2010.zip",
                    "02_00500-00625_HITEMP2010.zip",
                    "02_00625-00750_HITEMP2010.zip",
                    "02_00750-01000_HITEMP2010.zip",
                    "02_01000-01500_HITEMP2010.zip",
                    "02_01500-02000_HITEMP2010.zip",
                    "02_02000-02125_HITEMP2010.zip",
                    "02_02125-02250_HITEMP2010.zip",
                    "02_02250-02500_HITEMP2010.zip",
                    "02_02500-03000_HITEMP2010.zip",
                    "02_03000-03250_HITEMP2010.zip",
                    "02_03250-03500_HITEMP2010.zip",
                    "02_03500-03750_HITEMP2010.zip",
                    "02_03750-04000_HITEMP2010.zip",
                    "02_04000-04500_HITEMP2010.zip",
                    "02_04500-05000_HITEMP2010.zip",
                    "02_05000-05500_HITEMP2010.zip",
                    "02_05500-06000_HITEMP2010.zip"
                    ]
            else:
                inputfiles = [ # fix here, add all zips YUSUF CTRL F
                    "01_00000-00050_HITEMP2010.zip",
                    "01_00050-00150_HITEMP2010.zip",
                    "01_00150-00250_HITEMP2010.zip",
                    "01_00250-00350_HITEMP2010.zip",
                    "01_00350-00500_HITEMP2010.zip",
                    "01_00500-00600_HITEMP2010.zip",
                    "01_00600-00700_HITEMP2010.zip",
                    "01_00700-00800_HITEMP2010.zip",
                    "01_00800-00900_HITEMP2010.zip",
                    "01_00900-01000_HITEMP2010.zip",
                    "01_01000-01150_HITEMP2010.zip",
                    "01_01150-01300_HITEMP2010.zip",
                    "01_01300-01500_HITEMP2010.zip",
                    "01_01500-01750_HITEMP2010.zip",
                    "01_01750-02000_HITEMP2010.zip",
                    "01_02000-02250_HITEMP2010.zip",
                    "01_02250-02500_HITEMP2010.zip",
                    "01_02500-02750_HITEMP2010.zip",
                    "01_02750-03000_HITEMP2010.zip",
                    "01_03000-03250_HITEMP2010.zip",
                    "01_03250-03500_HITEMP2010.zip",
                    "01_03500-04150_HITEMP2010.zip",
                    "01_04150-04500_HITEMP2010.zip",
                    "01_04500-05000_HITEMP2010.zip",
                    "01_05000-05500_HITEMP2010.zip",
                    "01_05500-06000_HITEMP2010.zip",
                    "01_06000-06500_HITEMP2010.zip"
                    ]
                
            urlnames = [join(base_url, f) for f in inputfiles]
```

With these changes I could access and converted the files that I needed to the .h5 format that allows me to access the data and make the calculations I need.
